sentry-core/src/main/java/io/sentry/core/SentryCommonOptions.java

```diff
@@ -0,0 +1,263 @@
+package io.sentry.core;
+
+import com.jakewharton.nopen.annotation.Open;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Subset of {@link SentryOptions} properties used in 3rd party framework integrations. */
+@Open
+public class SentryCommonOptions {
+
+  /** @see SentryOptions#dsn */
+  private String dsn = "";
+
+  /** @see SentryOptions#shutdownTimeoutMillis */
+  private Long shutdownTimeoutMillis;
+
+  /** @see SentryOptions#flushTimeoutMillis */
+  private Long flushTimeoutMillis;
+
+  /** @see SentryOptions#readTimeoutMillis */
+  private Integer readTimeoutMillis;
+
+  /** @see SentryOptions#bypassSecurity */
+  private Boolean bypassSecurity;
+
+  /** @see SentryOptions#debug */
+  private Boolean debug;
+
+  /** @see SentryOptions#diagnosticLevel */
+  private SentryLevel diagnosticLevel = SentryLevel.DEBUG;
```

Comment:
can you use `SentryOptions.DEFAULT_DIAGNOSTIC_LEVEL` here?

Reply:
It should be null by default. My bad.

Reply:
It should be `DEBUG` by default (perhaps with a const `DEFAULT_DIAGNOSTIC_LEVEL`) though, right?

If the user sets `option.setDebug(true)` it should automatically start printing all diag msgs of all levels